src/index.ts

```diff
@@ -854,6 +870,57 @@ export abstract class PinejsClientCore<PinejsClient> {
 		}
 	}
 
+	private canRetryDefaultHandler(err: any) {
+		const code = err?.statusCode;
+		return code == null || code === 429 || (code >= 500 && code < 600);
+	}
+
+	protected async callWithRetry<T>(
+		fnCall: () => Promise<T>,
+		retry: RetryParameters,
+	): Promise<T> {
+		if (!retry) {
+			return await fnCall();
+		}
+
+		const defaultRetry = this.retry || {};
+		const minDelayMs = retry.minDelayMs ?? defaultRetry.minDelayMs ?? 0;
+		const maxDelayMs = retry.maxDelayMs ?? defaultRetry.maxDelayMs ?? 0;
+		const maxAttempts = retry.maxAttempts ?? defaultRetry.maxAttempts ?? 0;
+
+		if (
+			minDelayMs <= 0 ||
+			maxDelayMs <= 0 ||
+			maxAttempts <= 0 ||
+			minDelayMs > maxDelayMs
+		) {
+			throw new Error('pinejs-client invalid retry parameters');
+		}
```

Comment:
If you can separate these out slightly to give better error messages it'll make it much more useful, eg
```suggestion
		if (minDelayMs <= 0) {
			throw new Error('minDelayMs must be positive');
		}
		if (maxDelayMs<= 0) {
			throw new Error('maxDelayMs must be positive');
		}
		if (maxAttempts <= 0) {
			throw new Error('maxAttempts be positive');
		}
		if (minDelayMs > maxDelayMs) {
			throw new Error('maxDelayMs must be greater than or equal to minDelayMs');
		}
```

I know it adds code but it makes the ergonomics/error messages much better